### ISSUE TITLE
subprocess.CalledProcessError returned non-zero exit status 255.

### ISSUE THREAD

**Issue by RhettRautsaw** [biowdl/singularity-permanent-cache#5]
I have installed and setup a directory for a permanent cache. 
```
pip install singularity-permanent-cache
mkdir singularity_permanent_cache
export SINGULARITY_PERMANENTCACHEDIR=$PWD/singularity_permanent_cache
```

I can then run:
```
spc -vvv docker://gcr.io/deepvariant-docker/deepvariant:1.5.0
```
and it successfully generates `docker_gcr.io_deepvariant-docker_deepvariant_1.5.0.sif` in my `SINGULARITY_PERMANENTCACHEDIR`

However, I have since tried to do the same on other docker URIs and they all fail with the same message:
```
> spc -vvv docker://quay.io/pacbio/pbmm2@sha256:1013aa0fd5fb42c607d78bfe3ec3d19e7781ad3aa337bf84d144c61ed7d51fa1

INFO:root:Cache dir from environment: $SINGULARITY_PERMANENTCACHEDIR
INFO:root:Waiting for file lock on: $SINGULARITY_PERMANENTCACHEDIR/.lock
DEBUG:root:Lock acquired: $SINGULARITY_PERMANENTCACHEDIR/.lock
INFO:root:Start pulling image docker://quay.io/pacbio/pbmm2@sha256:1013aa0fd5fb42c607d78bfe3ec3d19e7781ad3aa337bf84d144c61ed7d51fa1 to location $SINGULARITY_PERMANENTCACHEDIR/docker_quay.io_pacbio_pbmm2@sha256_1013aa0fd5fb42c607d78bfe3ec3d19e7781ad3aa337bf84d144c61ed7d51fa1.sif
DEBUG:root:Lock released: $SINGULARITY_PERMANENTCACHEDIR/.lock
Traceback (most recent call last):
  File "/home/UNIXHOME/rrautsaw/miniconda3/envs/bio/bin/spc", line 8, in <module>
    sys.exit(main())
  File "/home/UNIXHOME/rrautsaw/miniconda3/envs/bio/lib/python3.10/site-packages/singularity_permanent_cache/singularity_permanent_cache.py", line 227, in main
    image_path = pull_image_to_cache(args.uri, cache_dir,
  File "/home/UNIXHOME/rrautsaw/miniconda3/envs/bio/lib/python3.10/site-packages/singularity_permanent_cache/singularity_permanent_cache.py", line 213, in pull_image_to_cache
    singularity_command(singularity_exe, "pull", str(image_tmp), uri)
  File "/home/UNIXHOME/rrautsaw/miniconda3/envs/bio/lib/python3.10/site-packages/singularity_permanent_cache/singularity_permanent_cache.py", line 156, in singularity_command
    result = subprocess.run([singularity_exe] + list(args),
  File "/home/UNIXHOME/rrautsaw/miniconda3/envs/bio/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['singularity', 'pull', '$SINGULARITY_PERMANENTCACHEDIR/docker_quay.io_pacbio_pbmm2@sha256_1013aa0fd5fb42c607d78bfe3ec3d19e7781ad3aa337bf84d144c61ed7d51fa1.tmp', 'docker://quay.io/pacbio/pbmm2@sha256:1013aa0fd5fb42c607d78bfe3ec3d19e7781ad3aa337bf84d144c61ed7d51fa1']' returned non-zero exit status 255.
```

**Comment by rhpvorderman** [comment]
This looks like an issue with singularity. Can you execute the command directly?

Also it seems like the program is using `$SINGULARITY_PERMANENTCACHEDIR` as a literal rather than pointing to an actual directory.

**Comment by RhettRautsaw** [comment]
I could run `singularity pull docker_quay.io_pacbio_pbmm2@sha256_1013aa0fd5fb42c607d78bfe3ec3d19e7781ad3aa337bf84d144c61ed7d51fa1.tmp docker://quay.io/pacbio/pbmm2@sha256:1013aa0fd5fb42c607d78bfe3ec3d19e7781ad3aa337bf84d144c61ed7d51fa1` without any issue.

And the error code is actually pointing to a path...I actually replaced it with `$SINGULARITY_PERMANENTCACHEDIR` above just to anonymize my file system and make the error more legible and less cluttered. 

That being said...having made 0 changes...and running the exact same command I ran the other day...it's working. So must have been something with singularity or my internet access at the time I guess....
